### ISSUE TITLE
color maps

### ISSUE THREAD

**Issue by CPernet** [neurolabusc/MRIcron#9]
Dear Chris, 

like many, I love using your tools (mricron, mricronGL, surfice) BUT have you considered linearizing the luminance of the colour maps? As you likely know this is important to have linear change in luminance along with colours to ensure good visualization. I'm finishing to write a paper on data viz including something on colour maps and I 'fixed' the colour maps, see here https://github.com/CPernet/brain_colours. (also for renders, isoluminant map might be a better option)

Note that the .lut do not work for mricron (??) - I'd be happy to help if you wanted to consider those new maps (also not sure what the clut format is in mricroGL)

best
cyril

**Comment by neurolabusc** [comment]
I have sent a direct email. Below is the code to read/write not only the ImageJ/MRIcron format LUT but also the FSLeyes format cmap files. Users can get [additional color tables here](https://imagej.nih.gov/ij/download/luts/). As an aside, I like the idea of the 'cube helix' color scheme, though I think it should be adapted to exploit the fact that the human eye is most sensitive at detecting changes in hue in the range between green and red.

The CLUT format is a very simple text format - the main feature is that it stores alpha (transparency) as well as color, and interpolates color between an arbitrary number of nodes. If you choose Color/ColorEditor you can graphically adjust these nodes, and having a few nodes makes it much easier to adjust than hundreds of nodes.

```
function lut = loadLutSub(fnm)
%load color lookup table in imageJ/MRIcron lut format or FSLeyes cmap format
if ~exist(fnm,'file'), error('Unable to find %s', fnm); end
stats = dir(fnm);
if stats.bytes == 768 %ImageJ/MRIcron format bytes 0..255 RRR..RGGG..GBBB..B
    fid = fopen(fnm,'rb');
    lut = fread(fid);
    fclose(fid);
    lut = reshape(lut,256,3);
    lut = lut / 255; %scale 0..255 -> 0..1
    return;
end
[~,~,x] = fileparts(fnm);
if strcmpi(x,'.lut'), lut = []; fprintf('Unable to read %s\n', fnm); return; end
fid = fopen(fnm,'r');
lut = fscanf(fid,'%f %f %f');
fclose(fid);
lut = reshape(lut,3,numel(lut)/3)';
%loadLutSub()
if (max(lut(:)) > 1.0) ||  (min(lut(:)) < 0), error('RGB should be in range 0..1 %s', fnm); end
if (size(lut,1) == 255)
    lut = [[0 0 0];lut];
end
if (size(lut,1) < 255)
    %interpolate
    lut = [[0 0 0];lut];
    R = lut(:,1);
    G = lut(:,1);
    B = lut(:,1);
    xq = 1: (numel(R)-1)/255 : numel(R);
    R = interp1(R,xq);
    G = interp1(G,xq);
    B = interp1(B,xq);
    lut = [R' G' B'];
end
%loadLutSub()

function saveCMapSub(pth, prefix, fnm, lut)
[~,n] = fileparts(fnm);
fid = fopen(fullfile(pth,[prefix, n,'.cmap']),'w');
fprintf(fid,'%0.6f %0.6f %0.6f\n',lut);
fclose(fid);
%end saveCMapSub()

function saveLutSub(pth, prefix, fnm, lut)
[~,n] = fileparts(fnm);
fid = fopen(fullfile(pth,[prefix, n,'.lut']),'wb');
fwrite(fid,lut*255,'uchar');
fclose(fid);
%end saveLutSub()

```

**Comment by CPernet** [comment]
Great thx, I have regenerated the .lut (also made .cmap) and it now works nicely simply substituting .lut files. I also added two diverging maps (blue-white-red and blue-grey-yellow) which are fully symmetric (unless eg NIH) -- for oceanographic maps it works beautifully (especially bgy), seems to also work for brains (played a bit to make green-red but not that easy). So if you wanted to change them, that's that simple.
